pyomo/core/util.py

```diff
@@ -66,6 +66,13 @@ def quicksum(args, start=0, linear=None):
     Returns:
         The value of the sum, which may be a Pyomo expression object.
     """
+
+    # Ensure that args is an iterator (this manages things like IndexedComponent_slice objects)
+    try:
+        args = iter(args)
+    except:
+        raise TypeError('The argument `args` to quicksum() is not iterable!')
```

Comment:
Instead of swallowing the `TypeError` and raising another, I think I would prefer that we log an error and re-raise (this makes the stack trace that the user sees less confusing by avoiding the "During handling of the above exception another exception occurred" message):
```python
try:
    args = iter(args)
except:
    logger.error('The argument `args` to quicksum() is not iterable!')
    raise
```
(and at the top of the module)
```python
import logging
logger = logging.getLogger(__name__)
```